### PR TITLE
chore: use cache for yarn

### PR DESCRIPTION
**Pull request by Jolg42**
And random cleanup

### CODE REVIEW

File: .github/workflows/check-for-update.yaml [hunk 1]
```diff
@@ -17,8 +17,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: use node 14
-        uses: actions/setup-node@v3
+      - uses: actions/setup-node@v3
```
Comment (janpio) [inline .github/workflows/check-for-update.yaml:20]: version being used still explicit in the logs?

File: .github/workflows/invoke-platforms-lambda-function.yml [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
-      
+
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
-      - run: yarn
+        with:
+          cache: 'yarn'
+      - name: Install Dependencies
+        run: yarn install
```
Comment (janpio) [inline .github/workflows/invoke-platforms-lambda-function.yml:25]: difference between just `yarn` and `yarn install`?

Comment (Jolg42) [reply .github/workflows/invoke-platforms-lambda-function.yml:25]: Just having one explicit everywhere, it's the same but easier to search

File: .github/workflows/optional-test.yaml [hunk 3]
```diff
@@ -33,6 +33,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
+        with:
+          cache: 'yarn'
```
Comment (janpio) [inline .github/workflows/optional-test.yaml:37]: Is this lacking an install deps step?

**Review by janpio — COMMENTED** [review]
Can we somehow extract these super common and repeated steps now maybe? I have some vagues recollection of GH Actions adding some composability recently for theri yaml.

File: .github/workflows/check-for-update.yaml (hunk 1 above)
Comment (Jolg42) [reply .github/workflows/check-for-update.yaml:20]: Only if you open the section. But not needed and can drift.
For tests where we explicitly want to test the node version we use a matrix and it's visible

File: .github/workflows/optional-test.yaml (hunk 3 above)
Comment (Jolg42) [reply .github/workflows/optional-test.yaml:37]: Apparently no yarn install is needed here because it has no dependencies, so cache is not needed as well

**Comment by Jolg42** [comment]
We can check, it could definitely help but not sure it's what we want here (the feature might not do what we want)

**Pull request merged**